src/entitytypes.js

```diff
@@ -976,6 +976,35 @@ SlugIdArray.prototype.indexOf = function(slug) {
   return -1;
 };
 
+/** Determines whether it includes a certain element, returning true or false as appropriate. */
+SlugIdArray.prototype.includes = function(slug) {
+  return this.buffer.includes(slugIdToBuffer(slug));
+};
+
+/** The shift() method removes element. */
+SlugIdArray.prototype.shift = function() {
+  if (this.length === 0) {
+    return;
+  }
+
+  this.buffer.copy(this.buffer, 0, SLUGID_SIZE);
```

Comment:
I have no idea what you're doing here... You probably need `length * SLUGID_SIZE`

Notice that `SlugIdArray` is designed to gracefully allow doubling of allocated buffer size, such that you get an asymptotic complexity of `O(1)` for `.push()` and `pop()`...
That won't be the case of this operation... I suggest we make a warning about this in the method docs.

Reply:
@jonasfj, the `shift` method removes the first element by shifting everything left by one.

`this.buffer.copy(this.buffer, 0, SLUGID_SIZE)` begins copying to offset 0 within the buffer but starts copying from offset `SLUGID_SIZE`. The 4th parameter (sourceEnd) is using the default value: 

`sourceEnd <integer> The offset within buf at which to stop copying (not inclusive). Ignored when sourceStart is undefined. Default: buf.length`.

[https://nodejs.org/api/buffer.html#buffer_buf_copy_target_targetstart_sourcestart_sourceend](https://nodejs.org/api/buffer.html#buffer_buf_copy_target_targetstart_sourcestart_sourceend)

I changed the method signature to:

```
/**
 * The shift() method removes the first element. Each operation will take a
 * time proportional to the number of the array length */
SlugIdArray.prototype.shift = function () {
  ...
};
```